reactor-netty-http/src/main/java/reactor/netty/http/server/NonSslRedirectDetector.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2011-Present VMware, Inc. or its affiliates, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package reactor.netty.http.server;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelPipeline;
+import io.netty.handler.codec.ByteToMessageDecoder;
+import io.netty.handler.ssl.SslHandler;
+import reactor.netty.NettyPipeline;
+import reactor.netty.tcp.SslProvider;
+
+import javax.annotation.Nullable;
+import java.net.SocketAddress;
+import java.util.List;
+
+/**
+ * @author James Chen
```

Comment:
Add `@since 1.0.5`

Reply:
Done.

Reply:
Please, fill out the Javadoc.

Reply:
@JamesChenX Can you add javadoc as per @rstoyanchev review comment

Reply:
@violetagg updated. ptal